### ISSUE TITLE
nokogiri-1.12.5 cannot be found

### ISSUE THREAD

**Issue by TrevorHinesley** [boltops-tools/jets#594]
## Checklist

- [x] Upgrade Jets: Are you using the latest version of Jets? This allows Jets to fix issues fast. There's a `jets upgrade` command that makes this a simple task. There's also an Upgrading Guide: http://rubyonjets.com/docs/upgrading/
- [ ] Reproducibility: Are you reporting a bug others will be able to reproduce and not asking a question. If you're unsure or want to ask a question, do so on https://community.rubyonjets.com
- [ ] Code sample: Have you put together a code sample to reproduce the issue and make it available?  Code samples help speed up fixes dramatically. If it's an easily reproducible issue, then code samples are not needed. If you're unsure, please include a code sample.

## My Environment

| Software         | Version |
| ---------------- | ------- |
| Operating System |  OS X 10.15.7 |
| Jets             | 3.0.17 |
| Ruby             | 2.7.0 |

---

## Expected Behaviour

Upon properly upgrading runtime from Ruby 2.5 to Ruby 2.7, function should be able to operate just fine

## Current Behavior

The exact error from CloudWatch Logs:

```json
{
    "errorMessage": "Could not find nokogiri-1.12.5 in any of the sources",
    "errorType": "Init<Bundler::GemNotFound>",
    "stackTrace": [
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/definition.rb:496:in `materialize'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/definition.rb:234:in `specs_for'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/runtime.rb:18:in `setup'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler.rb:149:in `setup'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/setup.rb:20:in `block in <top (required)>'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/ui/shell.rb:136:in `with_level'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/ui/shell.rb:88:in `silence'",
        "/var/runtime/gems/bundler-2.2.27/lib/bundler/setup.rb:20:in `<top (required)>'",
        "/var/lang/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
        "/var/lang/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
        "/var/task/handlers/jobs/upload_job.rb:1:in `<top (required)>'",
        "/var/lang/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
        "/var/lang/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'"
    ]
}
```

## Step-by-step reproduction instructions

My gemfile is very straightforward:

```ruby
source "https://rubygems.org"

gem "jets",                "~> 3.0.0"

gem "aws-sdk-s3",          "~> 1.51.0"
gem "content_disposition", "~> 1.0.0"
gem "dynamoid",            "~> 3.7.0"
gem "httparty",            "~> 0.16.2"

# development and test groups are not bundled as part of the deployment
group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
end

group :test do
  gem 'rspec' # rspec test group only or we get the "irb: warn: can't alias context from irb_context warning" when starting jets console
end

```

I was actually able to get the deployment to work one time using Bundler 2.2.27 and Ruby 2.7.0 to match Amazon's installed versions exactly, but any deployment before and after that one time, using the same versions, is causing the same issue. Very bizarre, but I'm thinking the one time it worked might've been a fluke (I was toying around with a forked version of Jets and changing some things to diagnose -- will test again).

## Code Sample

Unfortunately I have not had time to spin up a new instance, but will try to add that tomorrow if possible.

## Solution Suggestion

It seems as if https://github.com/boltops-tools/jets/issues/523 and https://github.com/boltops-tools/jets/pull/524 point to this being fixed, but I can't seem to resolve it. I imagine it's something to do with the same thing though?

**Comment by TrevorHinesley** [comment]
Actually, this appears to affect the Ruby 2.5 runtime as well. For some reason, I can't relaunch my 2.5 runtimes either. Same nokogiri issue. I've tried declaring it in my Gemfile as well.

**Comment by TrevorHinesley** [comment]
I found the problem here. My `Gemfile.lock` looked like this:

```
PLATFORMS
  x86_64-darwin-18
  x86_64-darwin-19
```

So this line was converting that to:

```
PLATFORMS
  ruby
  x86_64-darwin-19
```
 
 Manually removing the `x86_64-darwin-19` line before deploying fixed this. I'll create a PR to resolve this.

**Comment by TrevorHinesley** [comment]
The above PR should resolve this issue once merged.

**Comment by dbelling** [comment]
I've encountered a similar issue on my M1 mac. Would love to see the above Pull Request get merged into the mainline.

Additional system metadata if it is helpful:
| Software      | Version |
| ----------- | ----------- |
| Ruby      | ruby 2.7.2p137       |
| Jets   | 3.0.22        |
| Operating System | 11.6 (Big Sur) |
| rbenv | 1.2.0 |

**Comment by tongueroo** [comment]
Hope to get to soon. If you got time, just need to QA it. A sanity check with the fork and making sure jets deploy is good enough. No sweat either way.

**Comment by dbelling** [comment]
@tongueroo I spent some time with @esmarkowski today testing the solution proposed in #611 - I can confirm that the changes in that pull request address the underlying problem with platform-specific dependency linking.

**Comment by anthonygharvey** [comment]
@tongueroo I can also confirm that @esmarkowski, @dbelling and I were able to successfully deploy to AWS using the changes in PR #611.

**Comment by tongueroo** [comment]
@esmarkowski @dbelling anthonygharvey You all rock! Thanks for working together on this. Released in v3.0.23

**Comment by esmarkowski** [comment]
> @esmarkowski @dbelling @anthonygharvey You all rock! Thanks for working together on this. Released in v3.0.23@

Thanks @tongueroo!

**Comment by TrevorHinesley** [comment]
Great work everyone! Thanks for moving this along.
